### PR TITLE
SONAR-9616 Write main branch to DB even if plugin is not installed

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentUpdater.java [hunk 1]
```diff
@@ -100,6 +103,17 @@ private ComponentDto createRootComponent(DbSession session, NewComponent newComp
       .setPrivate(newComponent.isPrivate())
       .setCreatedAt(new Date(system2.now()));
     dbClient.componentDao().insert(session, component);
+
+    BranchDto branch = new BranchDto()
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentUpdater.java]: Warning, this class is used by Governance to create View.
And we should not create main branch on views.

I think the creation of the main branch should only be done during the analysis, not during the provisioning.

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterDelegate.java [hunk 2]
```diff
@@ -19,12 +19,96 @@
...
+import org.sonar.server.computation.task.projectanalysis.analysis.AnalysisMetadataHolder;
+import org.sonar.server.computation.task.projectanalysis.analysis.Branch;
 
-@ComputeEngineSide
-public interface BranchPersisterDelegate {
+import static org.sonar.db.component.ComponentDto.UUID_PATH_OF_ROOT;
+import static org.sonar.db.component.ComponentDto.UUID_PATH_SEPARATOR;
 
-  void persist(DbSession dbSession);
+public class BranchPersisterDelegate {
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterDelegate.java]: The name of the class should be changed, as it's no more a delegate

Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterDelegate.java]: => BranchPersister

Comment (dbmeneses) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterDelegate.java]: It is still a delegate, as much as it was before

File: server/sonar-server/src/main/java/org/sonar/server/component/ComponentUpdater.java (hunk 1 above)
Comment (dbmeneses) [inline server/sonar-server/src/main/java/org/sonar/server/component/ComponentUpdater.java]: it needs to be created as soon as the project is created if we want it to always be consistent. I will change it to only be done when it's a regular project.

File: server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterDelegate.java (hunk 2 above)
Comment (dbmeneses) [inline server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/component/BranchPersisterDelegate.java]: That being said, I agree with the name simplification :)

**Pull request closed without merging**